File: frameworks/dotnet/minimal-apis/Implementation/Routes/JsonRoutes.cs
namespace Implementation.Routes;

/// <summary>json: a payload the framework already holds, serialised at three sizes.</summary>
public static class JsonRoutes
{
    public static void Map(IEndpointRouteBuilder app)
    {
        app.MapGet("/json/small", (Payloads p) => p.Small);

        app.MapGet("/json/medium", (Payloads p) => p.Medium);

        app.MapGet("/json/large", (Payloads p) => p.Large);
    }
}

File: frameworks/java/vertx/Implementation/src/main/java/implementation/routes/JsonRoutes.java
package implementation.routes;

import implementation.Payloads;
import io.vertx.ext.web.Router;

/**
 * json: a payload the framework already holds, serialised at three sizes by RoutingContext.json,
 * which encodes a JsonObject with Jackson's streaming generator.
 */
public final class JsonRoutes {

    private JsonRoutes() {}

    public static void register(Router router, Payloads p) {
        router.get("/json/small").handler(ctx -> ctx.json(p.small()));

        router.get("/json/medium").handler(ctx -> ctx.json(p.medium()));

        router.get("/json/large").handler(ctx -> ctx.json(p.large()));
    }
}
